pkg/commands/workflows/plan_status_comments.go

```diff
@@ -132,95 +108,46 @@ func (c *PlanStatusCommentCommand) Run(ctx context.Context, args []string) error
 		return flag.ErrHelp
 	}
 
-	c.actions = githubactions.New(githubactions.WithWriter(c.Stdout()))
-	actionsCtx, err := c.actions.Context()
+	rc, err := reporter.NewReporter(ctx, c.flagReporter, &reporter.Config{GitHub: c.githubConfig}, c.Stdout())
 	if err != nil {
-		return fmt.Errorf("failed to load github context: %w", err)
+		return fmt.Errorf("failed to create reporter client: %w", err)
 	}
-
-	c.cfg = &PlanStatusCommentsConfig{}
-	if err := c.cfg.MapGitHubContext(actionsCtx); err != nil {
-		return fmt.Errorf("failed to load github context: %w", err)
-	}
-	logger.DebugContext(ctx, "loaded configuration", "plan_status_comments_config", c.cfg)
-
-	tokenSource, err := c.GitHubFlags.TokenSource(ctx, map[string]string{
-		"pull_requests": "write",
-	})
-	if err != nil {
-		return fmt.Errorf("failed to get token source: %w", err)
-	}
-	c.gitHubClient = github.NewClient(
-		ctx,
-		tokenSource,
-		github.WithRetryInitialDelay(c.RetryFlags.FlagRetryInitialDelay),
-		github.WithRetryMaxAttempts(c.RetryFlags.FlagRetryMaxAttempts),
-		github.WithRetryMaxDelay(c.RetryFlags.FlagRetryMaxDelay),
-	)
+	c.reporterClient = rc
 
 	return c.Process(ctx)
 }
 
 // Process handles the main logic for the Guardian remove plan comments process.
 func (c *PlanStatusCommentCommand) Process(ctx context.Context) error {
-	logger := logging.FromContext(ctx).
-		With("github_owner", c.GitHubFlags.FlagGitHubOwner).
-		With("github_repo", c.GitHubFlags.FlagGitHubOwner).
-		With("init_result", c.flagInitResult).
-		With("plan_result", c.flagPlanResult)
-
-	logger.DebugContext(ctx, "determining plan status...")
-
-	c.gitHubLogURL = fmt.Sprintf("[[logs](%s/%s/%s/actions/runs/%d/attempts/%d)]", c.cfg.ServerURL, c.GitHubFlags.FlagGitHubOwner, c.GitHubFlags.FlagGitHubRepo, c.cfg.RunID, c.cfg.RunAttempt)
-	logger.DebugContext(ctx, "computed github log url", "github_log_url", c.gitHubLogURL)
-
 	// this case improves user experience, when the planning job does not have a plan diff
 	// there are no comments, this informs the user that the job ran successfully
 	if c.flagInitResult == "success" && util.SliceContainsOnly(c.flagPlanResult, "success") {
```

Comment:
The string "success" is hardcoded twice here, should it become a constant?

Reply:
this is fair, added